include/xtensor/xstrides.hpp

```diff
@@ -169,17 +171,57 @@ namespace xt
 
     namespace detail
     {
+        template <typename Arg, typename... Args>
+        struct last_type_helper {
+            using type = typename last_type_helper<Args...>::type;
+        };
+
+        template <typename Arg>
+        struct last_type_helper<Arg> {
+            using type = Arg;
+        };
+
+        template <typename... Args>
+        using last_type = typename last_type_helper<Args...>::type;
+
+        template <typename... Args>
+        constexpr bool last_type_is_missing = std::is_same<missing_type, last_type<Args...>>::value;
```

Comment:
The `xtl` provides the [back](https://github.com/xtensor-stack/xtl/blob/master/include/xtl/xmeta_utils.hpp#L362) metafunction that you could directly use instead.

Reply:
I'm not sure how:
```cpp
 error: pack expansion used as argument for non-pack parameter of alias template
        constexpr bool last_type_is_missing = std::is_same<missing_type, xtl::mpl::back_t<Args...>>::value;
```

Reply:
I think yuo need to wrap the parameter pack into an `mpl::vector` before passing it to `back_t`:

```cpp
namespace mpl = xtl::mpl;
mpl::back_t<mpl::vector<Args...>>;
```